### ISSUE TITLE
access exported interface from import * as 

### ISSUE THREAD

**Issue by sophister** [typescript-eslint/typescript-eslint#589]
**What code were you trying to parse?**

```ts
// in file common.ts
export interface User {
  name: string;
  age: number;
};

export interface Profile {
  addr: string;
  city: string;
};

export enum Status {
  OK,
  Fail
};

// in another file  foo.ts
import * as all from './common.ts';
all.User;      // does not exist
all.Profile;   //does not exist
all.Status;  //OK
```

**What did you expect to happen?**
I can access `interface` with `all.User` and `all.Profile`
**What actually happened?**
`import * as ` lost exported `interface`
**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.7.0` |
| `TypeScript`                | `3.4.3` |
| `ESLint`                    | `5.16.0` |
| `node`                      | `10.14.1` |
| `npm`                       | `6.4.1` |

**Comment by bradzacher** [comment]
It looks like you've posted this to the wrong repo!
https://github.com/Microsoft/TypeScript/

**Comment by sophister** [comment]
@bradzacher  thx for your reply. But typescript itself can handle this case. I've written a demo here: 
[https://codesandbox.io/s/tsexportinterfaceall-8ro9k](https://codesandbox.io/s/tsexportinterfaceall-8ro9k)  . But when I use `typescript-eslint`, `interface` does not exist in the `all` .

**Comment by bradzacher** [comment]
I don't quite understand what the problem is?

What is the error message you're seeing?
That demo doesn't show any specific errors with the eslint parsing, or the eslint rules.

You're saying that the interfaces do not exist on `all`, but in your demo they clearly do?

![image](https://user-images.githubusercontent.com/7462525/58925820-cca93580-86fd-11e9-9a90-35884bc7d55b.png)

Additionally - `@typescript-eslint/parser` does not deal with typechecking - all it does is takes the code you write and converts it to an AST.

Please let me know the issue you have found with `@typescript-eslint/parser`.

**Comment by sophister** [comment]
@bradzacher I just found that, using `all.Status` is ok, I just can't use javascript object spread syntax with `all`.   So it has nothing to do with `typescript-eslint`.  **sorry** 

```javascript
// this does not work
const { Status, Profile } = all;
// this is valid
all.Status
all.Profile
```
![image](https://user-images.githubusercontent.com/219501/58944490-87940a80-87b4-11e9-94dc-a37cae3e0fcf.png)
